### DISCUSSION TITLE
Is it possible to combine `Union` and `OneOfObject`?

### DISCUSSION THREAD

**Discussion by teenjuna** [async-graphql/async-graphql#956] [Q&A]
I want to make an enum both `Union` and `OneOfObject`. When I combine them inside of `derive(...)` it compiles fine, but fails during the execution:
```
thread 'main' panicked at 'Register `Details` as `InputObject`, but it is already registered as `Union`'
```
I'm not familiar with the implementation details, but shouldn't it be possible to do something like `#[graphql(one_of_name = "...")]` to avoid such collision?

**Comment by binarybox** [comment]
From the issues it seems like it is working for structs 
https://github.com/async-graphql/async-graphql/issues/1116
